tests/lib/rules/no-useless-flag.ts

```diff
@@ -393,6 +467,85 @@ tester.run("no-useless-flag", rule as any, {
                 },
             ],
         },
+
+        // y
+        {
+            code: `
+            /* ✓ GOOD */
+            const regex1 = /foo/y;
+            const str = 'table football, foosball';
+            regex1.lastIndex = 6
+            var array = regex1.exec(str)
+            
+            const regex2 = /foo/y;
+            regex2.test(string);
+            regex2.test(string);
+            
+            str.replace(/foo/y, 'bar');
+            str.replaceAll(/foo/gy, 'bar');
+
+            const regexp3 = /foo/y
+            str.search(regexp3)
+            
+            /* ✗ BAD */
+            str.split(/foo/y);
+            `,
+            output: null,
+            errors: [
+                {
+                    message:
+                        "The 'y' flag is unnecessary because not using sticky search.",
+                    line: 19,
+                    column: 28,
+                },
+            ],
+        },
+        {
+            code: `
+            "str".split(/foo/y)
+            `,
+            output: null,
+            errors: [
+                {
+                    message:
+                        "The 'y' flag is unnecessary because not using sticky search.",
+                    line: 2,
+                    column: 30,
+                },
+            ],
+        },
+        {
+            code: `
+            "str".split(new RegExp('foo', 'y'));
+            `,
+            output: null,
+            errors: [
+                {
+                    message:
+                        "The 'y' flag is unnecessary because not using sticky search.",
+                    line: 2,
+                    column: 43,
+                },
+            ],
+        },
+        {
+            code: `
+            const a = /foo/y;
+            const b = a;
+            const regex = b;
+
+            'str'.split(b)
```

Comment:
Is this really correct?

I thought the default source mode is "script" which means that `a`, `b`, and `regex` are all global variables and might be used in another script where the `y` might be relevant.

Reply:
I searched for it. ESLint seems to judge the variable declared by the exported directive as a global variable.

https://eslint.org/docs/rules/no-unused-vars#exported

We should support `exported` too, but the problem is not limited to `no-useless-flag`. If `getUsageOfPattern` is also assigned to the exported variable, it should return `unknown`.
I would like to work on this issue as a separate issue.